File: exercise/75/solution.cpp
class Solution {
public:
    void swap(vector<int>& nums, int a, int b) {
        if (a == b) return;
        int t = nums[a];
        nums[a] = nums[b];
        nums[b] = t;
    }
    void sortColors(vector<int>& nums) {
        if (nums.size() == 0) {
            return;
        }
        int zero_end = 0, one_end = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 0) {
                swap(nums, zero_end, one_end);
                swap(nums, zero_end, i);
                zero_end++, one_end++;
            } else if (nums[i] == 1) {
                swap(nums, one_end, i);
                one_end++;
            }
        }
    }
};

File: exercise/75/solution.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        one, two = 0, 0
        for now in range(len(nums)):
            if nums[now] == 0:
                nums[two], nums[now] = nums[now], nums[two]
                nums[two], nums[one] = nums[one], nums[two]
                one += 1
                two += 1
            elif nums[now] == 1:
                nums[two], nums[now] = nums[now], nums[two]
                two += 1
            else:
                pass

